File: mobile/ios/MultimodalTrading/Views/Components/WalletView.swift
import SwiftUI

struct WalletView: View {
    @EnvironmentObject private var tradingViewModel: TradingViewModel
    @EnvironmentObject private var authViewModel: AuthViewModel
    
    var body: some View {
        VStack(spacing: 16) {
            Text("Wallet")
                .font(.headline)
                .frame(maxWidth: .infinity, alignment: .leading)
            
            if !authViewModel.isAuthenticated {
                // Not connected state
                VStack(spacing: 12) {
                    Image(systemName: "wallet.pass")
                        .font(.system(size: 36))
                        .foregroundColor(.gray)
                        .padding(.bottom, 8)
                    
                    Text("Connect your wallet to view your balance and transactions")
                        .font(.subheadline)
                        .foregroundColor(.gray)
                        .multilineTextAlignment(.center)
                    
                    Button(action: {
                        authViewModel.connect()
                    }) {
                        Text("Connect Wallet")
                            .font(.headline)
                            .foregroundColor(.white)
                            .frame(maxWidth: .infinity)
                            .padding()
                            .background(Color(hex: "4F46E5"))
                            .cornerRadius(12)
                    }
                }
                .padding()
            } else {
                // Connected state - show balances
                VStack(spacing: 16) {
                    // Token balances
                    ForEach(tradingViewModel.balances.sorted(by: { $0.key < $1.key }), id: \.key) { token, balance in
                        HStack {
                            // Token icon
                            ZStack {
                                Circle()
                                    .fill(tokenColor(token))
                                    .frame(width: 36, height: 36)
                                
                                Text(String(token.prefix(1)))
                                    .font(.system(size: 16, weight: .bold))
                                    .foregroundColor(.white)
                            }
                            
                            Text(token)
                                .font(.headline)
                            
                            Spacer()
                            
                            Text(balance)
                                .font(.headline)
                        }
                        .padding(.vertical, 4)
                    }
                    
                    Divider()
                        .padding(.vertical, 8)
                    
                    // Recent transactions
                    VStack(alignment: .leading, spacing: 12) {
                        Text("Recent Transactions")
                            .font(.subheadline)
                            .fontWeight(.medium)
                        
                        if tradingViewModel.orderHistory.isEmpty {
                            Text("No recent transactions")
                                .font(.subheadline)
                                .foregroundColor(.gray)
                                .padding(.vertical, 8)
                        } else {
                            ForEach(tradingViewModel.orderHistory.prefix(5)) { transaction in
                                VStack(spacing: 4) {
                                    HStack {
                                        Text(formatTransactionType(transaction))
                                            .font(.subheadline)
                                        
                                        Spacer()
                                        
                                        Text(transaction.timestamp)
                                            .font(.caption)
                                            .foregroundColor(.gray)
                                    }
                                    
                                    Divider()
                                }
                            }
                        }
                    }
                }
            }
        }
        .padding()
        .onAppear {
            if authViewModel.isAuthenticated {
                tradingViewModel.loadBalances()
                tradingViewModel.loadOrderHistory()
            }
        }
    }
    
    // Format transaction type
    private func formatTransactionType(_ transaction: Transaction) -> String {
        switch transaction.type {
        case "swap":
            return "Swap \(transaction.amount) \(transaction.token) to \(transaction.toAmount) \(transaction.toToken)"
        case "buy":
            return "Buy \(transaction.amount) \(transaction.token)"
        case "sell":
            return "Sell \(transaction.amount) \(transaction.token)"
        default:
            return "\(transaction.type.capitalized) \(transaction.amount) \(transaction.token)"
        }
    }
    
    // Get token color
    private func tokenColor(_ token: String) -> Color {
        switch token {
        case "SOL":
            return Color.blue
        case "USDC":
            return Color.green
        case "USDT":
            return Color(hex: "26A17B")
        default:
            return Color.gray
        }
    }
}

struct WalletView_Previews: PreviewProvider {
    static var previews: some View {
        WalletView()
            .environmentObject(TradingViewModel())
            .environmentObject(AuthViewModel())
            .previewLayout(.sizeThatFits)
            .padding()
    }
}

File: mobile/android/app/src/main/java/com/multimodaltrading/android/ui/components/WalletView.kt
package com.multimodaltrading.android.ui.components

import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.multimodaltrading.android.viewmodels.AuthViewModel
import com.multimodaltrading.android.viewmodels.TradingViewModel

@Composable
fun WalletView(
    tradingViewModel: TradingViewModel,
    modifier: Modifier = Modifier,
    authViewModel: AuthViewModel = androidx.lifecycle.viewmodel.compose.viewModel()
) {
    val isAuthenticated by authViewModel.isAuthenticated.collectAsState()
    val balances by tradingViewModel.balances.collectAsState()
    val orderHistory by tradingViewModel.orderHistory.collectAsState()
    
    Card(
        modifier = modifier
    ) {
        Column(
            modifier = Modifier.padding(16.dp)
        ) {
            Text(
                text = "Wallet",
                style = MaterialTheme.typography.titleMedium,
                modifier = Modifier.padding(bottom = 16.dp)
            )
            
            if (!isAuthenticated) {
                // Not connected state
                Column(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalAlignment = androidx.compose.ui.Alignment.CenterHorizontally
                ) {
                    Icon(
                        imageVector = androidx.compose.material.icons.Icons.Filled.AccountBalanceWallet,
                        contentDescription = "Wallet",
                        modifier = Modifier.size(48.dp),
                        tint = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                    
                    Spacer(modifier = Modifier.height(16.dp))
                    
                    Text(
                        text = "Connect your wallet to view your balance and transactions",
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.onSurfaceVariant,
                        modifier = Modifier.padding(horizontal = 16.dp)
                    )
                    
                    Spacer(modifier = Modifier.height(16.dp))
                    
                    Button(
                        onClick = { authViewModel.connect() }
                    ) {
                        Text("Connect Wallet")
                    }
                }
            } else {
                // Connected state - show balances
                Column(
                    modifier = Modifier.fillMaxWidth()
                ) {
                    // Token balances
                    balances.entries.sortedBy { it.key }.forEach { (token, balance) ->
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(vertical = 8.dp),
                            verticalAlignment = androidx.compose.ui.Alignment.CenterVertically
                        ) {
                            // Token icon
                            Surface(
                                modifier = Modifier.size(36.dp),
                                shape = MaterialTheme.shapes.extraLarge,
                                color = getTokenColor(token)
                            ) {
                                Box(
                                    contentAlignment = androidx.compose.ui.Alignment.Center
                                ) {
                                    Text(
                                        text = token.first().toString(),
                                        style = MaterialTheme.typography.titleMedium,
                                        color = MaterialTheme.colorScheme.onPrimary
                                    )
                                }
                            }
                            
                            Spacer(modifier = Modifier.width(16.dp))
                            
                            Text(
                                text = token,
                                style = MaterialTheme.typography.titleMedium
                            )
                            
                            Spacer(modifier = Modifier.weight(1f))
                            
                            Text(
                                text = balance,
                                style = MaterialTheme.typography.titleMedium
                            )
                        }
                    }
                    
                    Divider(modifier = Modifier.padding(vertical = 16.dp))
                    
                    // Recent transactions
                    Text(
                        text = "Recent Transactions",
                        style = MaterialTheme.typography.titleSmall,
                        modifier = Modifier.padding(bottom = 8.dp)
                    )
                    
                    if (orderHistory.isEmpty()) {
                        Text(
                            text = "No recent transactions",
                            style = MaterialTheme.typography.bodyMedium,
                            color = MaterialTheme.colorScheme.onSurfaceVariant,
                            modifier = Modifier.padding(vertical = 8.dp)
                        )
                    } else {
                        orderHistory.take(5).forEach { transaction ->
                            Column(
                                modifier = Modifier.padding(vertical = 8.dp)
                            ) {
                                Row(
                                    modifier = Modifier.fillMaxWidth(),
                                    horizontalArrangement = Arrangement.SpaceBetween
                                ) {
                                    Text(
                                        text = formatTransactionType(transaction),
                                        style = MaterialTheme.typography.bodyMedium
                                    )
                                    
                                    Text(
                                        text = transaction.timestamp,
                                        style = MaterialTheme.typography.bodySmall,
                                        color = MaterialTheme.colorScheme.onSurfaceVariant
                                    )
                                }
                                
                                Divider(modifier = Modifier.padding(top = 8.dp))
                            }
                        }
                    }
                }
            }
        }
    }
}

// Format transaction type
private fun formatTransactionType(transaction: com.multimodaltrading.android.models.Transaction): String {
    return when {
        transaction.type == "swap" && transaction.toToken != null -> 
            "Swap ${transaction.amount} ${transaction.token} to ${transaction.toAmount} ${transaction.toToken}"
        else -> "${transaction.type.capitalize()} ${transaction.amount} ${transaction.token}"
    }
}

// Get token color
private fun getTokenColor(token: String): androidx.compose.ui.graphics.Color {
    return when (token) {
        "SOL" -> androidx.compose.ui.graphics.Color.Blue
        "USDC" -> androidx.compose.ui.graphics.Color.Green
        "USDT" -> androidx.compose.ui.graphics.Color(0xFF26A17B)
        else -> androidx.compose.ui.graphics.Color.Gray
    }
}

// Extension function to capitalize first letter
private fun String.capitalize(): String {
    return this.replaceFirstChar { if (it.isLowerCase()) it.titlecase() else it.toString() }
}
